src/main.rs

```diff
@@ -224,13 +230,17 @@ async fn host_ip_provider_server(port: u16) -> Result<()> {
     }
 }
 
-async fn listen_and_serve(args: &Cli) -> Result<()> {
-    let host_listener = TcpListener::bind(&args.source_address)
+async fn listen_and_serve(
+    source_address: String,
+    destination_address: String,
+    args: Cli,
+) -> Result<()> {
+    let host_listener = TcpListener::bind(&source_address)
         .await
         .context("failed to start source listener")?;
     let conn_count_semaphore = Arc::new(Semaphore::new(args.max_concurrent_connections));
```

Comment:
So each proxy entry has an independent connection limit? Please update the help on `max_concurrent_connections` to mention this.